src/main/java/Duke.java

```diff
@@ -1,10 +1,214 @@
+    private static final String DELETEMESSAGE = "Noted. I've removed this task:";
+
+    // Commands
+    private static final String ENDCOMMAND = "bye";
+    private static final String PRINTCOMMAND = "list";
+    private static final String MARKCOMMAND = "mark";
+    private static final String UNMARKCOMMAND = "unmark";
+    private static final String TODOCOMMAND = "todo";
+    private static final String DEADLINECOMMAND = "deadline";
+    private static final String EVENTCOMMAND = "event";
+    private static final String DELETECOMMAND = "delete";
+
+    // Data structures
+    private static final List<Task> list = new ArrayList<>();
+
+    /**
+     * Adds param str to a List.
+     * @param str
+     */
+    private static void createTask(String str) {
+        Task newTask = new Task(str);
+        list.add(newTask);
+        prettyPrint(String.format("added: %s", newTask.getDescription()));
+    }
+
+    private static void createTask(String str, TaskType type){
+        try {
+
+            Task newTask;
+            switch(type) {
+                case TODO:
+                    if (str.strip().equals("")) {
+                        throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                    }
+                    newTask = new ToDo(str);
+                    break;
+                case DEADLINE:
+                    if (str.strip().equals("")) {
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    String[] strList = str.split(" /by ");
+                    newTask = new Deadline(strList[0], strList[1]);
+                    break;
+                case EVENT:
+                    if (str.strip().equals("")) {
+                        throw new DukeException("☹ OOPS!!! The description of an event cannot be empty.");
+                    }
+                    strList = str.split(" /at ");
+                    newTask = new Event(strList[0], strList[1]);
+                    break;
+                default:
+                    newTask = new Task(str);
+            }
+            list.add(newTask);
+            String singulStr = "Now you have 1 task in the list";
+            if (list.size() == 1) {
+                prettyPrint(String.format("%s\n%s   %s\n%s %s",
+                        CREATEMESSAGE, TAB, newTask.toString(), TAB, singulStr));
+            } else {
+                prettyPrint(String.format("%s\n%s   %s\n%s Now you have %d tasks in the list.",
+                        CREATEMESSAGE, TAB, newTask.toString(), TAB, list.size()));
+            }
+        } catch (DukeException e) {
+            prettyPrint(e.toString());
+        }
+    }
+
+    /**
+     * Marks a task at index as done
+     */
+    private static void markTask(String index) {
+        index = index.strip();
+        int i = Integer.parseInt(index);
```

Comment:
I'm not sure if a change here is required, but consider the following requirements:
`Variables named j, k etc. should be used for nested loops only.`